llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py

```diff
@@ -291,13 +291,29 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
         else:
             where = kwargs.pop("where", {})
 
-        results = self._collection.query(
+        if not query.query_embedding:
+            return self._get(limit=query.similarity_top_k, where=where, **kwargs)
```

Comment:
Should there still be a limit if its purely a filter query?

Reply:
I think so, from the perspective of the end-user they don't know whether the query() or get() method gets used, so when they set `smililarity_top_k` they expect a max of this number of results.